core/impl/src/main/java/org/jboss/jca/core/connectionmanager/pool/mcp/SemaphoreConcurrentLinkedDequeManagedConnectionPool.java

```diff
@@ -559,36 +584,45 @@ else if (debug)
 
                if (t instanceof ResourceException)
                {
-                  throw (ResourceException)t;
+                  throw (ResourceException) t;
                }
                else
                {
                   throw new ResourceException(
-                     bundle.unexpectedThrowableWhileTryingCreateConnection(
-                             clw != null ? clw.getConnectionListener() : null), t);
+                      bundle.unexpectedThrowableWhileTryingCreateConnection(
+                          clw != null ? clw.getConnectionListener() : null), t);
                }
             }
-         } 
-         else 
+         }
+         else
          {
             if (pool.getInternalStatistics().isEnabled())
                pool.getInternalStatistics().deltaBlockingFailureCount();
 
             // We timed out
             throw new ResourceException(
-               bundle.noMManagedConnectionsAvailableWithinConfiguredBlockingTimeout(
-                  poolConfiguration.getBlockingTimeout()));
+                bundle.noMManagedConnectionsAvailableWithinConfiguredBlockingTimeout(
+                    poolConfiguration.getBlockingTimeout()));
          }
 
-      } 
-      catch (InterruptedException ie) 
+      }
+      catch (InterruptedException ie)
       {
          Thread.interrupted();
```

Comment:
It is generally incorrect to swallow interrupts in this manner.
If Condition.await() throws an InterruptedException, the interrupt status of the thread is reset (i.e. Thread.interrupted() will return false).

> If the current thread:
> 
> has its interrupted status set on entry to this method; or
> is interrupted while waiting and interruption of thread suspension is supported,
> then InterruptedException is thrown and the current thread's interrupted status is cleared. It is not specified, in the first case, whether or not the test for interruption occurs before the lock is released.

If any of the waiting operations in this method throw an InterruptedException, the interrupt status of the calling thread should reflect this upon method return.  To do this, you should handle the InterruptedException by _setting_ the interrupt status of the thread.  
e.g.
```java
try {
    // ...
    condition.await();
    // ...
} catch (InterruptedException e) {
    Thread.currentThread().interrupt();
    // ...
}
```

Reply:
fixed